modification_request.go

```diff
@@ -0,0 +1,66 @@
+package braintree
+
+import (
+	"encoding/xml"
+)
+
+type ModificationRequest struct {
+	Amount                *Decimal `xml:"amount,omitempty"`
+	NumberOfBillingCycles int      `xml:"number-of-billing-cycles,omitempty"`
+	Quantity              int      `xml:"quantity,omitempty"`
+	NeverExpires          bool     `xml:"never-expires,omitempty"`
+}
+
+type AddModificationRequest struct {
+	ModificationRequest
+	InheritedFromID string `xml:"inherited-from-id,omitempty"`
+}
+
+type UpdateModificationRequest struct {
+	ModificationRequest
+	ExistingID string `xml:"existing-id,omitempty"`
+}
+
+type ModificationsRequest struct {
+	Add               []AddModificationRequest
+	Update            []UpdateModificationRequest
+	RemoveExistingIDs []string
+}
+
+func (m ModificationsRequest) MarshalXML(e *xml.Encoder, start xml.StartElement) error {
```

Comment:
@TamalSaha: Ah, that would be simpler since the xml representation would still be described using tags on structs rather than a custom marshaller. I'll give that a go and see how it looks. Thanks for the feedback!